refactor: Eliminating `ui.js` module.

Lalit3716 opened the pull request:
<!-- Describe your pull request here.-->
To completely eliminate `ui.js` module
This PR does 3 things:
- [x] Move update_message_lists related functions to `message_live_update`.
- [x] Create new `compose_textarea` module and extract related handlers from `ui.js`.
- [x] Merge remaining functions of `ui.js` into `ui_util` module.
<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

Lalit3716 commented:
@timabbott This is ready for review!

timabbott commented on web/src/ui_util.ts:
```diff
@@ -133,3 +133,16 @@ export function get_content_element($element: JQuery): JQuery {
+        return $(new SimpleBar(element).getScrollElement()!);
+    }
+    return $element;
+}
```
Hmm, looking at this more, I think it might be better to merge these into `web/src/scroll_util.js` rather than `ui_util.js` -- it's a more focused/specific module.

Lalit3716 replied:
Yeah `scroll_util` seems more appropriate module for them I will move them there.

Lalit3716 commented on web/src/message_live_update.js:
```diff
@@ -31,6 +31,56 @@ function rerender_messages_view_for_user(user_id) {
+    update_message_in_all_views(message_id, ($row) => {
+        $row.find(".message_failed").toggleClass("hide", true);
+    });
+}
```
@timabbott So should we keep `update_message_in_all_views` in `message_live_update` or move it as well in `echo.js`?

Lalit3716 replied:
Yeah I suppose it makes sense to move them into `echo.js` will do that.

Lalit3716 commented:
#### Updates:

- Moved `show_message_failed` and `show_failed_message_success` to `echo.js`
for cleaner separation of responsibilities.
- Moved all the simplebar helper functions to `scroll_util` instead of `ui_util`, also squashed those commits.

#### Question:

After these changes `scroll_util.test.js` needed to mock the helper function `scroll_util.get_scroll_element` but I could not mock the module which is being tested 😅, so for the workaround I decided to tweak the `get_scroll_element` function to work with the test by adding this if check:
```javascript
    // For testing we just return the element itself.
    if ($element && $element.__zjquery) {
        return $element;
    }
``` 
and then inside the test I added this property to the fake container:
```javascript
run_test("scroll_element_into_container", () => {
    const $container = (function () {
        let top = 3;
        return {
            height: () => 100,
            scrollTop: (arg) => {
                if (arg === undefined) {
                    return top;
                }
                top = arg;
                return this;
            },
            __zjquery: true,
        };
    })();
``` 
Let me know if there is anything better I can do here for testing "scroll_element_into_container" function.

Lalit3716 commented:
Sure I will update here when that PR gets merged.

timabbott commented:
That testing change looks reasonable, and I think #25224 might actually be nicer on top of this, so merging, thanks @Lalit3716!

If you can convert `scroll_util` to TS soon, I expect that may be required to complete #25224.

Lalit3716 commented:
Sure I will put up a PR soon to migrate `scroll_util`!
